src/pyuvdata/beam_interface.py

```diff
@@ -69,7 +77,7 @@ def __post_init__(self, include_cross_pols: bool):
         if isinstance(self.beam, UVBeam):
             if self.beam_type is None or self.beam_type == self.beam.beam_type:
                 self.beam_type = self.beam.beam_type
-            elif self.beam_type == "power":
+            elif self.beam_type == "power" and self.beam.beam_type != "power":
```

Comment:
This clause isn't required because it's guaranteed to be True. The previous `if` statement already catches the case where the beam_types are the same. Maybe add a comment instead if it's not clear?